tls/extensions/s2n_certificate_extensions.h

```diff
@@ -21,9 +21,13 @@
 
 /* Guards against errors and non uint16s, then increments size */
 #define GUARD_UINT16_AND_INCREMENT( x, size ) do { \
+    GUARD_UINT16(x); \
+    size += x; \
```

Comment:
Should this check for overflow?

Reply:
the result `total_size` gets GUARD_UINT16()-ed